WeatherApiApp/WeatherApiApp/ViewController/WeatherViewController.swift

```diff
@@ -9,25 +9,46 @@
 import UIKit
 import Alamofire
 import SwiftyJSON
+import CoreLocation // 位置情報を取得するために必要なモジュールをインポートする。
 
 class WeatherViewController: UIViewController {
     
-    @IBOutlet var max: UILabel!
-    @IBOutlet var min: UILabel!
-    @IBOutlet var taikan: UILabel!
-    @IBOutlet var humidity: UILabel!
-    @IBOutlet var wind: UILabel!
+    @IBOutlet private var max: UILabel!
+    @IBOutlet private var min: UILabel!
+    @IBOutlet private var taikan: UILabel!
+    @IBOutlet private var humidity: UILabel!
+    @IBOutlet private var wind: UILabel!
+    private var currentLocation: CLLocation?
+    // 位置情報を取得するために必要なプロトコルを付与する。
+    let locationManager = CLLocationManager()
     
     override func viewDidLoad() {
         super.viewDidLoad()
-        getWeatherData(zip: "950-0086", country: "JP")
+        self.locationManager.delegate = self // CLLocationManagerDelegateプロトコルを使用するためにdelegateをViewControllerクラスに設定する。
+        locationManager.requestWhenInUseAuthorization() // 位置情報の許可設定を通知する。
+    }
+    
+    @IBAction func getLocation(_ sender: Any) {
+        locationManager.requestLocation()
+    }
+}
+
+extension WeatherViewController: CLLocationManagerDelegate {
+    func locationManager(_ manager: CLLocationManager, didUpdateLocations locations: [CLLocation]) {
+        guard let loc = locations.last else { return }
+        currentLocation = loc
+        getWeatherData(latitude: String(currentLocation?.coordinate.latitude ?? 0), longitude: String(currentLocation?.coordinate.longitude ?? 0))
```

Comment:
currentLocation変数噛ませずとも、これだけでうまくいきそうです！
guard letの定数名をcurrentLocationにして、既存のcurrentLocationは消していただけると🙏

```suggestion
        getWeatherData(latitude: String(loc.coordinate.latitude), longitude: String(loc.coordinate.longitude))
```